File: uikit/src/main/java/io/agora/chat/uikit/chat/interfaces/OnRecallMessageResultListener.java
package io.agora.chat.uikit.chat.interfaces;


import io.agora.chat.ChatMessage;

public interface OnRecallMessageResultListener {
    /**
     * 撤回成功
     * @param message
     */
    void recallSuccess(ChatMessage message);

    /**
     * 撤回失败
     * @param code
     * @param errorMsg
     */
    void recallFail(int code, String errorMsg);
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/interfaces/OnRecallMessageResultListener.kt
package io.agora.uikit.feature.chat.interfaces

import io.agora.uikit.common.ChatMessage


interface OnRecallMessageResultListener {
    /**
     * Recall successful
     * @param originalMessage The message was unsent
     * @param notification  The notification message
     */
    fun recallSuccess(originalMessage: ChatMessage?, notification: ChatMessage?)

    /**
     * Recall failed
     * @param code
     * @param errorMsg
     */
    fun recallFail(code: Int, errorMsg: String?)
}